student_explorer/backends.py

```diff
@@ -1,18 +1,38 @@
 from djangosaml2.backends import Saml2Backend
 from django.core.exceptions import PermissionDenied
+from django.conf import settings
+
 import logging
 
 logger = logging.getLogger(__name__)
 
 
 class ActiveUserOnlySAML2Backend(Saml2Backend):
     def authenticate(self, **kwargs):
-        user = super(ActiveUserOnlySAML2Backend, self).authenticate(**kwargs)
+        user = None
+        try:
+            user = super(ActiveUserOnlySAML2Backend, self).authenticate(**kwargs)
+        except PermissionDenied:
+            # If user 
+            if user:
+                user.is_active = False
+                user.save()
+            raise PermissionDenied
+        # If the user doesn't exist for some reason return permission denied
         if not user:
             raise PermissionDenied
-        if user.is_active:
-           return user
+        # The user should be made active if they exist and aren't active
+        if not user.is_active:
+           user.is_active = True
+           user.save()
+        return user
+            
+
+    def is_authorized(self, attributes, attribute_mapping):
+        # If there are any groups that we're a member of, return true
+        if attributes.get('isMemberOf'):
+            logger.debug(attributes.get('isMemberOf'))
+            return True
         else:
-            logger.error(
-                'The user "%s" exists but is set inactive', user.username)
+            logger.warn('The user "%s" is not in one of the allowed groups', attributes.get('uid'))
```

Comment:
Can we also log the allowed groups info in the warning message?

Reply:
I think also we need to document somewhere how the allowed groups are passed into Shibb. And provide a sample Shibb response XML with the isMemberOf attribute.

Reply:
There is a debug message where the 'isMemberOf' is printed. I can have it printed as a warning in the exception instead of or in addition to the  debug. 

I'm not sure how this was defined, I think it was done through a special configuration by the Shibboleth team.

Reply:
So if it gets into this else case, it's because the user isn't in any groups. So there will just be an empty list